share/change.py

```diff
@@ -292,18 +292,30 @@ def change(self):
         # Hacky fix for SHARE-604
         # If the given date_updated is older than the current one, don't accept any changes that would overwrite newer changes
         ignore_attrs = set()
-        if issubclass(self.model, apps.get_model('share', 'creativework')) and 'date_updated' in self.attrs and self.instance.date_updated:
-            date_updated = pendulum.parse(self.attrs['date_updated'])
-            if date_updated < self.instance.date_updated:
-                logger.warning('%s appears to be from the past, change date_updated (%s) is older than the current (%s). Ignoring conflicting changes.', self, self.attrs['date_updated'], self.instance.date_updated)
-                # Just in case
-                ignore_attrs.update(self.instance.change.change.keys())
-
-                # Go back until we find a change that is older than us
-                for version in self.instance.versions.select_related('change').all():
-                    if not version.date_updated or date_updated > version.date_updated:
-                        break
-                    ignore_attrs.update(version.change.change.keys())
+        if issubclass(self.model, apps.get_model('share', 'creativework')) and self.instance:
+            if 'date_updated' in self.attrs and self.instance.date_updated:
+                date_updated = pendulum.parse(self.attrs['date_updated'])
+                if date_updated < self.instance.date_updated:
+                    logger.warning('%s appears to be from the past, change date_updated (%s) is older than the current (%s). Ignoring conflicting changes.', self, self.attrs['date_updated'], self.instance.date_updated)
+                    # Just in case
+                    ignore_attrs.update(self.instance.change.change.keys())
+
+                    # Go back until we find a change that is older than us
+                    for version in self.instance.versions.select_related('change').all():
+                        if not version.date_updated or date_updated > version.date_updated:
+                            break
+                        ignore_attrs.update(version.change.change.keys())
+
+            # Pretty self-explanatory. If we get changes from a source that hasn't been marked as canonical
+            # Don't allow attributes set by canonical sources to be changed.
+            # Stops aggregators from providng more generic sources or overwriting changes that have been pushed to use.
```

Comment:
> Stops aggregators from providng more generic sources or overwriting changes that have been pushed to use.

This sentence isn't clear. I'm not sure what "providing more generic sources" means or what "changes that have been pushed to use" means. Would "Stops aggregators from overwriting attributes set by canonical sources." cover the meaning? If so, we might not need the last sentence.